### PR TITLE
chore: add default gpo constants

### LINKED ISSUE #5661: Introduce constants for GasPriceOracleArgs default values
### Describe the feature

ref https://github.com/paradigmxyz/reth/pull/5660#discussion_r1412637142

Currently we have some `Default` values for `GasPriceOracleArgs`:
https://github.com/paradigmxyz/reth/blob/be68b780ee526dd8665f8b8744a89007ab66469a/bin/reth/src/args/gas_price_oracle_args.rs#L25-L32

These should be introduced as constants and used there

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yash-atreya**
Fixes #5661

### CODE REVIEW

File: crates/primitives/src/constants/mod.rs [hunk 1]
```diff
@@ -179,6 +179,19 @@ pub const BEACON_ROOTS_ADDRESS: Address = address!("000F3df6D732807Ef1319fB7B8bB
...
 pub const SYSTEM_ADDRESS: Address = address!("fffffffffffffffffffffffffffffffffffffffe");
 
+/// Number of recent blocks to check for gas price
+pub const DEFAULT_GAS_PRICE_BLOCKS: u32 = 20;
+
+/// Gas Price below which gpo will ignore transactions
+pub const DEFAULT_GAS_PRICE_IGNORE: u64 = 2;
+
+/// Maximum transaction priority fee(or gasprice before London Fork) to be recommended by gpo
+pub const DEFAULT_GAS_PRICE_MAX: u64 = 500_000_000_000;
```
Comment (Rjected) [inline crates/primitives/src/constants/mod.rs:189]:
```suggestion
/// Maximum transaction priority fee (or gas price before London Fork) to be recommended by the gas price oracle
pub const DEFAULT_GAS_PRICE_MAX: u64 = 500_000_000_000;
```

File: crates/primitives/src/constants/mod.rs [hunk 2]
```diff
@@ -179,6 +179,19 @@ pub const BEACON_ROOTS_ADDRESS: Address = address!("000F3df6D732807Ef1319fB7B8bB
...
+
+/// The percentile of gas prices to use for the estimate
+pub const DEFAULT_GAS_PRICE_PERCENTILE: u32 = 60;
+
+///
```
Comment (Rjected) [inline crates/primitives/src/constants/mod.rs:194]:
```suggestion
```

File: crates/primitives/src/constants/mod.rs [hunk 3]
```diff
@@ -179,6 +179,19 @@ pub const BEACON_ROOTS_ADDRESS: Address = address!("000F3df6D732807Ef1319fB7B8bB
 /// block.
 pub const SYSTEM_ADDRESS: Address = address!("fffffffffffffffffffffffffffffffffffffffe");
 
+/// Number of recent blocks to check for gas price
+pub const DEFAULT_GAS_PRICE_BLOCKS: u32 = 20;
+
+/// Gas Price below which gpo will ignore transactions
```
Comment (Rjected) [inline crates/primitives/src/constants/mod.rs:185]:
```suggestion
/// Gas Price below which the gas price oracle will ignore transactions
```

**Review by Rjected — CHANGES_REQUESTED** [review]
Just some doc comment suggestions, otherwise this lgtm

**Review by Rjected — APPROVED** [review]
looks good to me, thanks!

**Review by rkrasiuk — COMMENTED** [review]
should these be in primitives crate? iirc it has no notion of gpo, so [rpc constants](https://github.com/paradigmxyz/reth/blob/0d522e84729038e3ff50a57e1fc79f019e82a5b2/crates/rpc/rpc-builder/src/constants.rs) is a better place to put it

**Review by mattsse — CHANGES_REQUESTED** [review]
I think so too,
can thenbe reexported from rpc-builder

**Comment by yash-atreya** [comment]
> should these be in primitives crate? iirc it has no notion of gpo, so [rpc constants](https://github.com/paradigmxyz/reth/blob/0d522e84729038e3ff50a57e1fc79f019e82a5b2/crates/rpc/rpc-builder/src/constants.rs) is a better place to put it

It's done

**Review by onbjerg — APPROVED** [review]
lgtm

**Review by mattsse — CHANGES_REQUESTED** [review]
this needs a bit more work:

use constants here:

https://github.com/paradigmxyz/reth/blob/main/bin/reth/src/args/gas_price_oracle_args.rs#L7-L21

get rid of Options

unify constants:

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/gas_oracle.rs#L20-L26

remove new function and use default initializor (`GasPriceOracleConfig {..Default:default}`) instead where needed

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/gas_oracle.rs#L70-L70

**Comment by yash-atreya** [comment]
> this needs a bit more work:
> 
> use constants here:
> 
> https://github.com/paradigmxyz/reth/blob/main/bin/reth/src/args/gas_price_oracle_args.rs#L7-L21
> 
> get rid of Options
> 
> unify constants:
> 
> https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/gas_oracle.rs#L20-L26
> 
> remove new function and use default initializor (`GasPriceOracleConfig {..Default:default}`) instead where needed
> 
> https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/gas_oracle.rs#L70-L70

@mattsse Almost done here.

To unify constants, I need to import the constants from `rpc-builder` and remove them from https://github.com/paradigmxyz/reth/blob/227e1b7ad513977f4f48b18041df02686fca5f94/crates/rpc/rpc/src/eth/gas_oracle.rs#L20-L26

This will require me to add `reth-rpc-builder` in the Cargo.toml of `rpc` but this causes a cyclical dependency. 
Need some guidance here. How can I import the constants from rpc-builder without causing this problem?

File: bin/reth/src/args/rpc_server_args.rs [hunk 4]
```diff
@@ -362,12 +362,7 @@ impl RethRpcConfig for RpcServerArgs {
     }
 
     fn gas_price_oracle_config(&self) -> GasPriceOracleConfig {
-        GasPriceOracleConfig::new(
-            self.gas_price_oracle.blocks,
-            self.gas_price_oracle.ignore_price,
-            self.gas_price_oracle.max_price,
-            self.gas_price_oracle.percentile,
-        )
+        GasPriceOracleConfig::default()
```
Comment (mattsse) [inline bin/reth/src/args/rpc_server_args.rs:361]: here we need to use the values from self:

```
GasPriceOracleConfig {
  max_prise: self.max_price,
  ....
..Default::default
}

File: bin/reth/src/args/gas_price_oracle_args.rs [hunk 5]
```diff
@@ -1,24 +1,23 @@
 use clap::Args;
-
 /// Parameters to configure Gas Price Oracle
 #[derive(Debug, Clone, Args, PartialEq, Eq, Default)]
 #[clap(next_help_heading = "Gas Price Oracle")]
 pub struct GasPriceOracleArgs {
     /// Number of recent blocks to check for gas price
     #[arg(long = "gpo.blocks", default_value = "20")]
```
Comment (mattsse) [inline bin/reth/src/args/gas_price_oracle_args.rs:11]: use the actual constant for all arguments: `default_value_t = <const>`

**Review by mattsse — CHANGES_REQUESTED** [review]
almost there

**Review by mattsse — APPROVED** [review]
ty!

lgtm

**Pull request merged**